src/lib/dialog_ui.sh

```diff
@@ -517,6 +517,76 @@ function create_choice_list_screen()
   return "$ret"
 }
 
+function create_form_screen()
+{
+  local box_title="$1"
+  local message_box="$2"
+  local back_title="$3"
+  local -n _fields_list="$4"
+  local ok_label="$5"
+  local cancel_label="$6"
+  local extra_label="$7"
+  local height="$8"
+  local width="$9"
+  local flag="${10}"
+  local choice_description
+  local index=0
+  local cmd
+  local ret
+  local start_text_field=0
+  local auxiliar_label_size=0
+  local row=1
+
+  flag=${flag:-'SILENT'}
+  height=${height:-$DEFAULT_HEIGHT}
+  width=${width:-$DEFAULT_WIDTH}
+  back_title=${back_title:-"${KW_UPSTREAM_TITLE}"}
+  ok_label=${ok_label:-"$ok_label"}
+  cancel_label=${cancel_label:-"$cancel_label"}
+  extra_label=${extra_label:-"$extra_label"}
+
+  # Escape all single quotes to avoid breaking arguments
+  box_title=$(str_escape_single_quotes "$box_title")
+  message_box=$(str_escape_single_quotes "$message_box")
```

Comment:
Don't forget to escape single quotes in the label variables also. Although improbable, any label with an unescaped single quote can crash the feature.